operations/service_deploy_smoke_utils.go

```diff
@@ -68,7 +68,32 @@ func (tests smokeEndpointTestDefinitions) checkEndpoints() error {
 	return errors.Wrapf(catcher.Resolve(), "failed to get %d endpoints", catcher.Len())
 }
 
-func checkTaskByCommit(username, key, commit string) error {
+func getLatestGithubCommit() (string, error) {
+	client := util.GetHttpClient()
+	defer util.PutHttpClient(client)
+
+	resp, err := client.Get("https://api.github.com/repos/evergreen-ci/evergreen/git/refs/heads/master")
+	defer resp.Body.Close()
```

Comment:
Do this after err to avoid panic.

Reply:
Actually, that's still wrong. Do this right here:
```
if resp != nil {
        defer resp.Body.Close()
}
```

Reply:
Fixed. I think defer after err check if correct. The example from the docs is
```
resp, err := http.Get("http://example.com/")
if err != nil {
	// handle error
}
defer resp.Body.Close()
body, err := ioutil.ReadAll(resp.Body)
```
https://golang.org/pkg/net/http/